### ISSUE TITLE
Add a simple NaiveBaye model

### ISSUE THREAD

**Issue by pommedeterresautee** [wush978/FeatureHashing#80]
I understand that this package has a very specific purpose : **hashing of a matrix**.
It is not a general ML package at all, and should not try to mimick carret or scikit.
However it will be mainly used in ML context.

I was thinking to add a simple NaiveBayes function to the package.

The idea is to provide a very intuitive model, with very few parameter which can be used by many as a base line for their tests.

I have noticed that none of the existing package support `dgCMatrix` matrix and even if you convert it to spam format which is sometimes supported, the matrix is converted to a dense data.frame before computation!!!! So you almost always finish with a memory allocation error.

I have already written a basic code for sparse Matrix working well on `dgCMatrix` matrix.

Moreover this basic model may be used in the example instead of more advanced one like XGBoost.

Can you tell me if you want me to make a PR?

Kind regards,
Michael

**Comment by wush978** [comment]
@pommedeterresautee ,

I still do not want to include the functions which does not related to *hashing of a matrix*. 

However, there is a compromising way to put the NaiveBayes into the package. I did include a ML algorithm [FTPRL](https://github.com/wush978/FeatureHashing/blob/master/inst/ftprl.R) into this package, but it is not an official function for this package. It is for the vignette. 

So, I suggest you to implement the NaiveBayes like what I did for FTPRL and write a vignette for showing how to do analysis with both FeatureHashing and NaiveBayes. How do you think?

Best,
Wush

**Comment by pommedeterresautee** [comment]
My concern is that Vignette content is not well referenced by most R documentation search motor.
And results from these search motors are well referenced on Google.
So in some way few people would be aware.

Another solution would be to put the two algorithms in a kind of FeatureHashing "accessory" package like many packages do for datasets.

IMO Naive Bayes is too simple to deserve its own package but with FTPRL it would make sense...

However it means maintaining two packages, may be not that good idea.
Let me know what you think. 

Btw I have no strong feeling against putting NB in a Vignette text too :-) 

Kind regards,
Michael

**Comment by wush978** [comment]
I agree your point that the naivebayse is too simple to deserve its own package. However, writing them in vignette makes sense to me. For advance users, they can write their own implementation. For beginners, they need vignette to understand why and when they should use the naivebayse with FeatureHashing. 

I think we could put the implementation under `inst/` first. Once we collect many of them, we could write  a package for them.

Wush

**Comment by pommedeterresautee** [comment]
It sounds ok to me.
